.github/workflows/upload_all.yml

```diff
@@ -1,26 +1,132 @@
+        run: exit 1
+      - run: echo "Uploading ${{ github.ref }}"
+      - name: Set upload targets
+        run: |
+          if [[ ${{ github.event.inputs.targets }} =~ all|google-play ]]; then
+              echo "Uploading to Google Play"
+              echo "POLYPOD_UPLOAD_GOOGLE_PLAY=true" >> $GITHUB_ENV
+          fi
+          if [[ ${{ github.event.inputs.targets }} =~ all|app-store ]]; then
+              echo "Uploading to the App Store"
+              echo "POLYPOD_UPLOAD_APP_STORE=true" >> $GITHUB_ENV
+          fi
+      - uses: actions/checkout@v2
+
+      # API and features
+      - name: API and features - Setup
+        uses: actions/setup-node@v2
+        with:
+          node-version: 16
+          cache: "npm"
+          cache-dependency-path: "**/package-lock.json"
+      - name: API and features - Lint
+        run: ./build.js lint
+      - name: API and features - Build
+        run: ./build.js
+        env:
+          POLYPOD_POLYPEDIA_REPORT_URL: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_URL }}
+          POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION }}
+      - name: API and features - Test
+        run: ./build.js test
+
+      # Android
+      - name: Android - Setup
+        uses: actions/setup-java@v1
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        with:
+          java-version: 11
+      - name: Android - Test
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        working-directory: platform/android
+        run: ./gradlew test
+      - name: Android - Write POLYPOD_ANDROID_UPLOAD_KEYSTORE_FILE
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        working-directory: platform/android/app
+        run: 'echo -n "$KEYSTORE" | base64 --decode > keystore.tmp'
+        env:
+          KEYSTORE: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEYSTORE_FILE }}
+      - name: Android - Build
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        working-directory: platform/android
+        run: ./gradlew assemble -Psign=true
+        env:
+          POLYPOD_ANDROID_UPLOAD_KEYSTORE_FILE: keystore.tmp
+          POLYPOD_ANDROID_UPLOAD_KEYSTORE_PASS: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEYSTORE_PASS }}
+          POLYPOD_ANDROID_UPLOAD_KEY_ALIAS: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEY_ALIAS }}
+      - name: Android - Set VERSION_CODE environment variable.
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        run: echo "VERSION_CODE=$( ./getPackageInfo.sh /usr/local/lib/android/sdk/build-tools/30.0.0/aapt ./app/build/outputs/apk/release/app-release.apk versionCode )" >> $GITHUB_ENV
+        working-directory: platform/android
+      - name: Android - Set VERSION_NAME environment variable.
+        if: env.POLYPOD_UPLOAD_GOOGLE_PLAY == 'true'
+        run: echo "VERSION_NAME=$( ./getPackageInfo.sh /usr/local/lib/android/sdk/build-tools/30.0.0/aapt ./app/build/outputs/apk/release/app-release.apk versionName )" >> $GITHUB_ENV
```

Comment:
Also repeated from above. Either turn the whole thing into a shell variable, or at least define the version string somewhere else.